x/profiles/keeper/keeper_chain_links.go

```diff
@@ -95,3 +94,64 @@ func (k Keeper) DeleteAllUserChainLinks(ctx sdk.Context, user string) {
 		k.DeleteChainLink(ctx, link)
 	}
 }
+
+// getOldestUserChainByChain returns the oldest chain link of the given owner associated to the given chain name
+// If the such the chain link is not exists, return false instead.
+func (k Keeper) getOldestUserChainByChain(ctx sdk.Context, owner, chainName string) (types.ChainLink, bool) {
+	var oldestLink types.ChainLink
+	found := false
+	k.IterateUserChainLinksByChain(ctx, owner, chainName, func(link types.ChainLink) (stop bool) {
+		if !found {
+			oldestLink = link
+			found = true
+			return false
+		}
+
+		if link.CreationTime.Before(oldestLink.CreationTime) {
+			oldestLink = link
+		}
+		return false
+	})
+
+	return oldestLink, found
+}
+
+// updateOwnerDefaultExternalAddress updates the default external address of the given chain name for the given owner
+// It must be performed after deleting the default external address chain link
+func (k Keeper) updateOwnerDefaultExternalAddress(ctx sdk.Context, owner, chainName string) {
+	store := ctx.KVStore(k.storeKey)
+	link, found := k.getOldestUserChainByChain(ctx, owner, chainName)
+	if found {
+		srcAddrData, err := types.UnpackAddressData(k.cdc, link.Address)
+		if err != nil {
+			panic(err)
+		}
+		k.SaveDefaultExternalAddress(ctx, owner, chainName, srcAddrData.GetValue())
+		return
+	}
+
+	// If owner has no chain link on the given chain name, then delete the key
+	store.Delete(types.DefaultExternalAddressKey(owner, chainName))
+}
+
+// SaveDefaultExternalAddress stores the given address as a default external address
+func (k Keeper) SaveDefaultExternalAddress(ctx sdk.Context, owner, chainName, target string) {
+	store := ctx.KVStore(k.storeKey)
+	store.Set(types.DefaultExternalAddressKey(owner, chainName), []byte(target))
+}
+
+// HasDefaultExternalAddress tells whether the given owner has a default external address on the given chain name or not
+func (k Keeper) HasDefaultExternalAddress(ctx sdk.Context, owner, chainName string) bool {
+	store := ctx.KVStore(k.storeKey)
+	return store.Has(types.DefaultExternalAddressKey(owner, chainName))
+}
+
+// isDefaultExternalAddress tells whether the given chain link is a default external address or not
+func (k Keeper) isDefaultExternalAddress(ctx sdk.Context, link types.ChainLink) bool {
+	store := ctx.KVStore(k.storeKey)
+	if !k.HasDefaultExternalAddress(ctx, link.User, link.ChainConfig.Name) {
+		return false
+	}
```

Comment:
I think these lines can be removed safely, as `string([]byte(nil))` will be `nil` and thus not equals to the link address itself